src/libs/record/record.c

```diff
@@ -0,0 +1,780 @@
+ */
+bool elektraRecordRemoveKeys (KDB * handle, KeySet * toRemove, bool recursive, Key * errorKey)
+{
+	if (handle == NULL)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "NULL pointer passed for KDB handle");
+		return false;
+	}
+
+	if (toRemove == NULL)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "NULL pointer passed for keys to remove");
+		return false;
+	}
+
+	if (ksGetSize (toRemove) == 0)
+	{
+		// Nothing to do
+		return true;
+	}
+
+	Key * sessionRecordingKey = keyNew (ELEKTRA_RECORD_SESSION_KEY, KEY_END);
+	KeySet * recordStorage = ksNew (0, KS_END);
+
+	// Load data from session diff
+	if (kdbGet (handle, recordStorage, sessionRecordingKey) == -1)
+	{
+		elektraCopyError (errorKey, sessionRecordingKey);
+		keyDel (sessionRecordingKey);
+		ksDel (recordStorage);
+		return false;
+	}
+
+	ElektraDiff * sessionDiff = getDiffFromSessionStorage (recordStorage, NULL);
+
+	for (elektraCursor it = 0; it < ksGetSize (toRemove); it++)
+	{
+		Key * k = ksAtCursor (toRemove, it);
+
+		if (recursive == false)
+		{
+			elektraDiffRemoveKey (sessionDiff, k);
+		}
+		else
+		{
+			elektraDiffRemoveSameOrBelow (sessionDiff, k);
+		}
+	}
+
+	putDiffIntoSessionStorage (recordStorage, sessionDiff);
+
+	bool successful = true;
+
+	// store data for session diff
+	if (kdbSet (handle, recordStorage, sessionRecordingKey) == -1)
+	{
+		elektraCopyError (errorKey, sessionRecordingKey);
+		successful = false;
+	}
+
+	elektraDiffDel (sessionDiff);
+	ksDel (recordStorage);
+	keyDel (sessionRecordingKey);
+
+	return successful;
+}
+
+bool elektraRecordGetDiff (KDB * handle, ElektraDiff ** diff, Key * errorKey)
+{
+	if (handle == NULL)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "NULL pointer passed for KDB handle");
+		return false;
+	}
+
+	if (diff == NULL)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "NULL pointer passed for diff");
+		return false;
+	}
+
+	Key * sessionRecordingKey = keyNew (ELEKTRA_RECORD_SESSION_KEY, KEY_END);
+	KeySet * recordStorage = ksNew (0, KS_END);
+
+	// Load data from session diff
+	if (kdbGet (handle, recordStorage, sessionRecordingKey) == -1)
+	{
+		elektraCopyError (errorKey, sessionRecordingKey);
+		keyDel (sessionRecordingKey);
+		ksDel (recordStorage);
+		return false;
+	}
+
+	ElektraDiff * sessionDiff = getDiffFromSessionStorage (recordStorage, NULL);
+	*diff = sessionDiff;
+
+	keyDel (sessionRecordingKey);
+	ksDel (recordStorage);
+
+	return true;
+}
+
+static KeySet * buildExportKeySet (ElektraDiff * diff)
+{
+	KeySet * ks = ksNew (0, KS_END);
+
+	KeySet * addedKeys = elektraDiffGetAddedKeys (diff);
+	ksAppend (ks, addedKeys);
+	ksDel (addedKeys);
+
+	KeySet * modifiedNewKeys = elektraDiffGetModifiedNewKeys (diff);
+	ksAppend (ks, modifiedNewKeys);
+	ksDel (modifiedNewKeys);
+
+	KeySet * removedKeys = elektraDiffGetRemovedKeys (diff);
+	for (elektraCursor i = 0; i < ksGetSize (removedKeys); i++)
+	{
+		Key * k = ksAtCursor (removedKeys, i);
+		keySetMeta (k, "meta:/elektra/deleted", "true");
```

Comment:
```suggestion
		keySetMeta (k, "meta:/elektra/deleted", "1");
```

Elektra uses `1` for `true`

Reply:
sure